src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -239,22 +261,952 @@ private void ImportLoadFloat(double value)
 
         private void ImportShiftOperation(ILOpcode opcode)
         {
-            throw new NotImplementedException();
+            StackItem op1 = PopWithValidation();
+            StackItem op2 = PopWithValidation();
+
+            int shiftBy = default(int);
+
+            switch (op1.Kind)
+            {
+                case StackValueKind.Int32:
+                    shiftBy = op1.AsInt32();
+                    break;
+                case StackValueKind.Int64:
+                    shiftBy = (int)op1.AsInt64();
+                    break;
+                case StackValueKind.NativeInt:
+                    shiftBy = op1.AsNativeInt().ToInt32();
+                    break;
+                case StackValueKind.Float:
+                case StackValueKind.Unknown:
+                case StackValueKind.ByRef:
+                case StackValueKind.ObjRef:
+                case StackValueKind.ValueType:
+                default:
+                    ThrowHelper.ThrowInvalidProgramException();
+                    break;
+            }
+
+            switch (op2.Kind)
+            {
+                case StackValueKind.Int32:
+                    {
+                        int value = op2.AsInt32();
+                        if (opcode == ILOpcode.shl)
```

Comment:
I think the major reason I don't have switch for this is because there's already a surrounding switch, nested switches might be much harder to read. Alternatively, I could have an `else` block

Reply:
> nested switches might be much harder to read.

Nested anything is harder to read. I do not think that converting the inner one to use if checks instead of switch makes it easier to read.

Reply:
This been switched up to use switch